### PR TITLE
x-pack/filebeat/module/threatintel/malwarebazaar: properly handle code_sign array

### LINKED ISSUE #29972: [Filebeat] Malware Bazaar - mapping for code_sign conflicts with actual data
The mapping for the `[threatintel.malwarebazaar.code_sign` field is a `keyword` but the actual data appears to be an array of code signer objects.

https://github.com/elastic/beats/blob/febc7ddbfa4b8c7f39fb8a05b8d90864f0199e93/x-pack/filebeat/module/threatintel/malwarebazaar/_meta/fields.yml#L44-L45

### Pipeline Error

`
failed to parse field [threatintel.malwarebazaar.code_sign] of type [text] in document with id 'qYdkJH4Bnuhw9_gm5tD7'. Preview of field's value: '{subject_cn=iPhone Distribution: China Mobile Group Shandong Co., Ltd., issuer_cn=Apple Worldwide Developer Relations Certification Authority, valid_to=2024-05-06T06:07:31Z, thumbprint=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx, valid_from=2021-05-07T06:07:32Z, serial_number=ssssssssssssssssssssssssssssssss, thumbprint_algorithm=SHA256, algorithm=sha256WithRSAEncryption}'
`

### Sample data

Sample [678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79](https://bazaar.abuse.ch/sample/678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79/)

`curl -X POST -F 'query=get_info' -F 'hash=678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79' https://mb-api.abuse.ch/api/v1/ | jq . -S`

```json
{
  "data": [
    {
      "anonymous": 0,
      "code_sign": [
        {
          "algorithm": "sha256WithRSAEncryption",
          "cscb_listed": false,
          "cscb_reason": "",
          "issuer_cn": "Apple Worldwide Developer Relations Certification Authority",
          "serial_number": "7dd984a17e2bf0bddf54abe8cd583e42",
          "subject_cn": "iPhone Distribution: China Mobile Group Shandong Co., Ltd.",
          "thumbprint": "384fde6e3c3e8cc10ba16b948beba37536823aeab9847027782431ca94de78e2",
          "thumbprint_algorithm": "SHA256",
          "valid_from": "2021-05-07T06:07:32Z",
          "valid_to": "2024-05-06T06:07:31Z"
        }
      ],
      "comment": null,
      "comments": null,
      "delivery_method": null,
      "dhash_icon": "a261bae8d2a896ca",
      "file_information": [
        {
          "context": "cape",
          "value": "https://www.capesandbox.com/analysis/217234/"
        }
      ],
      "file_name": "678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79",
      "file_size": 688472,
      "file_type": "exe",
      "file_type_mime": "application/x-dosexec",
      "first_seen": "2022-01-04 09:03:45",
      "imphash": "163ce73e6369b6830f7ee781052d2470",
      "intelligence": {
        "clamav": [
          "SecuriteInfo.com.Trojan005246d51.14116.12142.UNOFFICIAL"
        ],
        "downloads": "157",
        "mail": null,
        "uploads": "2"
      },
      "last_seen": "2022-01-04 11:02:21",
      "md5_hash": "03970e27a3e5fbbffbbc249769f9f3bf",
      "ole_information": [],
      "origin_country": "IT",
      "reporter": "JAMESWT_MHT",
      "sha1_hash": "bf8bf87788034e67399584ea672031366b13db61",
      "sha256_hash": "678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79",
      "sha3_384_hash": "6159d988d477309048cf7e540a23ed9a7b6b4c42b45e3e192ce08ddf8cd89cf4325a49146e80e8bc33b299fe36f81c83",
      "signature": null,
      "ssdeep": "6144:x9ZURaJOfiCIorsoCv0h7E9cwFyNkkCIjq7O3IWU+MagLMNLeJFuxmXROnWnK6ia:x8Ry8ixorsoC8ZGVYd3nUrCfsXRzK6ia",
      "tags": [
        "42 192 232 209",
        "exe",
        "signed"
      ],
      "telfhash": null,
      "tlsh": "T196E48D06BED280FFD655193014BA67769A3A5B020B35CFC39398DF592D33171AE3A13A",
      "vendor_intel": {
        "ANY.RUN": [
          {
            "analysis_url": "https://app.any.run/tasks/f2b8db7d-ea40-4e18-9e47-90f650002f30",
            "date": "2022-01-04 09:05:21",
            "file_name": "678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79",
            "tags": [
              "installer"
            ],
            "verdict": "No threats detected"
          }
        ],
        "CAPE": {
          "detection": null,
          "link": "https://www.capesandbox.com/analysis/217234/"
        },
        "CERT-PL_MWDB": {
          "detection": null,
          "link": "https://mwdb.cert.pl/sample/678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79/"
        },
        "FileScan-IO": {
          "confidence": "1",
          "report_link": "https://www.filescan.io/uploads/61d40d8892bdc4b4077d88e8/reports/724ce288-3f55-420a-a48e-3426210b122c/overview",
          "threatlevel": "1",
          "verdict": "MALICIOUS"
        },
        "InQuest": {
          "details": [
            {
              "category": "info",
              "description": "Found a Windows Portable Executable (PE) binary. Depending on context, the presence of a binary is suspicious or malicious.",
              "title": "Windows PE Executable"
            }
          ],
          "url": null,
          "verdict": "SUSPICIOUS"
        },
        "Intezer": {
          "analysis_url": "https://analyze.intezer.com/analyses/2936ba72-6e2a-48cb-aeb9-1a3adcfdc6dd?utm_source=MalwareBazaar",
          "family_name": null,
          "verdict": "malicious"
        },
        "ReversingLabs": {
          "first_seen": "2021-11-17 17:43:00",
          "scanner_count": "28",
          "scanner_match": "17",
          "scanner_percent": "60.71",
          "status": "MALICIOUS",
          "threat_name": "Win32.Trojan.Emotet"
        },
        "Triage": {
          "link": "https://tria.ge/reports/220104-k1lq4sgaa3/",
          "malware_config": [],
          "malware_family": null,
          "score": "1",
          "signatures": [
            {
              "score": null,
              "signature": "Suspicious use of SetWindowsHookEx"
            }
          ],
          "tags": []
        },
        "UnpacMe": [
          {
            "detections": [],
            "link": "https://www.unpac.me/results/fd48e40e-5342-465c-bccd-97c6ed748ca9/",
            "md5_hash": "03970e27a3e5fbbffbbc249769f9f3bf",
            "sha1_hash": "bf8bf87788034e67399584ea672031366b13db61",
            "sha256_hash": "678f39a7f5bc61c4c9a0f34a0fe44f49fea5349cd51f9c8808ee2b6602ff1d79"
          }
        ],
        "YOROI_YOMI": {
          "detection": "Legit File",
          "score": "0.00"
        },
        "vxCube": {
          "behaviour": [
            {
              "rule": "Сreating synchronization primitives",
              "threat_level": "neutral"
            },
            {
              "rule": "Searching for synchronization primitives",
              "threat_level": "neutral"
            },
            {
              "rule": "Creating a window",
              "threat_level": "neutral"
            },
            {
              "rule": "DNS request",
              "threat_level": "neutral"
            }
          ],
          "maliciousness": "0",
          "verdict": "clean1"
        }
      },
      "yara_rules": null
    }
  ],
  "query_status": "ok"
}
```

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Previously, the code_sign field was treated as a keyword which broke handling of
the structured data in the code signing details. The handling of the code signing
also incorrectly assumed that only a single value could be present (in reality,
this is the case for all examples in the malwarebazaar data set, but may not be
in the future) by using rename rather than appending the details to the relevant
x509 fields. This change maps the code_sign field as a nested field and appends
into the x509 fields.

## Why is it important?

Previously data in the code_sign field would prevent documents from being processed.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #29972

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -54,7 +54,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 - Fix MISP documentation for `var.filters` config option. {pull}31434[31434]
 - Duplicate awscloudwatch.* fields to aws.cloudwatch.* in aws-cloudwatch input. {pull}31488[31488]
 - aws-s3 input: Stop SQS keep-alive routine on InvalidParameterValue error. {issue}30675[30675] {pull}31499[31499]
-- Supporting the double digit date parsing in ingest pipeline for oracle logs {pull}31514[31514]
+- Supporting the double digit date parsing in ingest pipeline for oracle logs. {pull}31514[31514]
+- Fix handling of code_sign data in ThreatIntel Malwarebazaar. {issue}29972[29972] {pull}[]
```
Comment (efd6) [inline CHANGELOG.next.asciidoc:58]:
```suggestion
- Fix handling of code_sign data in ThreatIntel Malwarebazaar. {issue}29972[29972] {pull}31552[31552]
```

File: x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json [hunk 2]
```diff
@@ -416,5 +416,184 @@
...
+        "threat.indicator.file.x509.not_before": [
+            "2021-05-07T06:07:32Z",
+            "2022-03-31T14:03:55Z"
+        ],
+        "threat.indicator.file.x509.public_key_algorithm": [
+            "sha1WithRSAEncryption",
+            "sha256WithRSAEncryption"
+        ],
+        "threat.indicator.file.x509.serial_number": [
+            "",
```
Comment (efd6) [inline x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json:586]: This empty string is here to test that the processor handles the unlikely cases where the code signing is missing fields. I attempted to conditionally not append when `_ingest?._value?.<fieldname> != ""` but this invariably broke the pipeline with timeouts (using a variety of candidate syntaxes).

Comment (adriansr) [reply x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json:586]: Unfortunately, it's not possible to access inside a condition `_ingest._value` as it is not visible to painless.

The only alternative I can suggest is to use a script processor to split the array of objects into separate array fields. But if we're not expecting missing fields then this approach is good

Comment (adriansr) [reply x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json:586]: For illustrative purposes:

```
  - script:
      lang: painless
      if: 'ctx.abusech?.malwarebazaar?.code_sign != null'
      source: |
        Map dest = new HashMap();
        for (def obj : ctx.abusech.malwarebazaar.code_sign) {
          for (def entry : obj.entrySet()) {
            dest.computeIfAbsent(entry.getKey(), k -> new ArrayList()).add(entry.getValue());
          }
        }
        ctx["_temp_"] = dest;

  - rename:
      field: _temp_.subject_cn
      target_field: threat.indicator.file.x509.subject.common_name
      ignore_missing: true
  - rename:
      field: _temp_.issuer_cn
      target_field: threat.indicator.file.x509.issuer.common_name
      ignore_missing: true
  - rename:
      field: _temp_.algorithm
      target_field: threat.indicator.file.x509.public_key_algorithm
      ignore_missing: true
  - rename:
      field: _temp_.valid_from
      target_field: threat.indicator.file.x509.not_before
      ignore_missing: true
  - rename:
      field: _temp_.valid_to
      target_field: threat.indicator.file.x509.not_after
      ignore_missing: true
  - rename:
      field: _temp_.serial_number
      target_field: threat.indicator.file.x509.serial_number
      ignore_missing: true
```

**Review by adriansr — APPROVED** [review]
LGTM

File: x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/module/threatintel/malwarebazaar/test/malwarebazaar.ndjson.log-expected.json:586]: Thanks. We can revisit this if we find that empty fields ever find their way into the `code_sign` array.

**Pull request merged**